### PR TITLE
feat: support enforceEx, fix #70

### LINKED ISSUE #70: need enforceEx function
php-casbin had enforce, could add enforceEx function ?

https://casbin.org/docs/en/management-api#enforceex

### PR DESCRIPTION
**Pull request by basakest**

### CODE REVIEW

**Comment by hsluoyz** [comment]
@techoner

File: src/CoreEnforcer.php [hunk 1]
```diff
@@ -623,13 +624,30 @@ protected function enforcing(string $matcher, ...$rvals): bool
...
+                $explains = $this->model['p']['p']->policy[$explainIndex];
+            }
+        }
 
         if (Log::getLogger()->isEnabled()) {
             $reqStr = 'Request: ';
             $reqStr .= implode(', ', array_values($rvals));
 
-            $reqStr .= sprintf(' ---> %s', (string)$result);
+            $reqStr .= sprintf("---> %s\n", (string)$result);
```
Comment (techoner) [inline src/CoreEnforcer.php:641]: Please keep spaces in the log:
```
$reqStr .= sprintf(" ---> %s\n", var_export($result, true);
```

File: src/CoreEnforcer.php [hunk 2]
```diff
@@ -623,13 +624,30 @@ protected function enforcing(string $matcher, ...$rvals): bool
...
+
+            $reqStr = 'Hit Policy: ';
+            foreach ($explains as $i => $pval) {
+                if ($i != count($explains) - 1) {
+                    $reqStr .= sprintf("%s, ", (string)$pval);
+                } else {
+                    $reqStr .= sprintf("%s \n", (string)$pval);
```
Comment (techoner) [inline src/CoreEnforcer.php:648]: Print the array like this :
```
$reqStr .= sprintf("%s \n", '['.implode(', ',$pval).']');
```

**Review by techoner — CHANGES_REQUESTED** [review]

**Pull request merged**